File: clients/web/js/renderer/block/image.js
import { registerBlock } from "./core.js";
import { createElement } from "../../dom.js";

registerBlock("image", (node) => {
    const img = createElement("img");

    img.classList.add("image");

    img.src = node.props.src;

    if (node.props.alt) {
        img.alt = node.props.alt;
    }

    return img;
});

File: clients/yakoon-client-console/src/yakoon/client/renderer/image.py
from .base import BaseRenderer


class ImageRenderer(BaseRenderer):

    def __init__(self, node):
        self.node = node

    def render(self) -> str:
        src = self.node.props.get("src")
        alt = self.node.props.get("alt") or "unnamed"
        return f"[Image: {alt} - {src}]\n"
